src/ucp/wireup/wireup.c

```diff
@@ -602,6 +606,61 @@ ucp_wireup_process_reply(ucp_worker_h worker, ucp_ep_h ep,
     }
 }
 
+static UCS_F_NOINLINE void
+ucp_wireup_process_ep_check(ucp_worker_h worker, ucp_ep_h ep,
+                            const ucp_wireup_msg_t *msg,
+                            const ucp_unpacked_address_t *remote_address)
+{
+    ucs_status_t status;
+    ucp_ep_h reply_ep;
+    unsigned addr_indices[UCP_MAX_LANES];
+    ucs_status_ptr_t req;
+
+    if (ep != NULL) {
+        return;
+    }
+
+    status = ucp_worker_create_ep(worker, 0, remote_address->name,
+                                  "wireup ep_check reply", &reply_ep);
+    if (status != UCS_OK) {
+        ucs_error("Failed to create EP: %s", ucs_status_string(status));
+        return;
+    }
+
+    /* Initialize lanes (possible destroy existing lanes) */
+    status = ucp_wireup_init_lanes_by_request(worker, reply_ep, 0,
+                                              remote_address, addr_indices);
+    if (status != UCS_OK) {
+        goto failed_destroy_ep;
+    }
+
+    ucp_ep_update_remote_id(reply_ep, msg->src_ep_id);
+    status = ucp_wireup_msg_send(reply_ep, UCP_WIREUP_MSG_EP_FAILED,
+                                 0, NULL);
+    if (status != UCS_OK) {
+        goto failed_destroy_ep;
+    }
+
+    req = ucp_ep_close_flushed(ep, &ucp_request_null_param);
+    if (UCS_PTR_IS_PTR(req)) {
+        ucp_request_free(req);
+    }
+
+out:
+    return;
+
+failed_destroy_ep:
+    ucp_ep_destroy_internal(reply_ep);
+    goto out;
+}
+
+static UCS_F_NOINLINE void
+ucp_wireup_process_ep_failed(ucp_worker_h worker, ucp_ep_h ep)
```

Comment:
why need this wrapper? Can just call `ucp_worker_set_ep_failed`

Reply:
the general way to process wireup messages... removed